### PR TITLE
4093 create vaccine course loaders

### LINKED ISSUE #4093: Create vaccine course loaders
## Is your feature request related to a problem? Please describe 👀

**Create Loaders**
- [x] demographic indicator Loader for ProgramDetailView (Vaccine Course List)
- [x] vaccine courses for Immunisation ProgramList 
- [x] Vaccine course items for vaccine course edit (VaccineCourseDetailView)
- [x] Vaccine Schedule for vaccine course edit (VaccineCourseDetailView)

**UI Changes**
- [x] Add list of vaccine Course Name to ImmunisationProgramList 
- [x] Add Demographic Indicator Name to VaccineCourseList

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #4093

# 👩🏻‍💻 What does this PR do?

Vaccine Course names now show in the Immunisation Programs List View

image

And Demographic (should) show in the vaccine course list view (we can't add demographic from the front end yet)
image

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [X] Should be tested as part of the epic

# 📃 Documentation

- [X] **Part of an epic**: documentation will be completed for the feature as a whole

### CODE REVIEW

File: server/graphql/demographic/src/types.rs [hunk 1]
```diff
@@ -121,56 +122,6 @@ impl DemographicProjectionConnector {
         }
     }
 }
-#[derive(PartialEq, Debug)]
```
Comment (jmbrunskill) [inline server/graphql/demographic/src/types.rs:124]: Moved the nodes to shared `types` crate.
Tried to move the minimum required, so connectors etc are still here.

File: server/service/src/vaccine_course/test/query.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #[cfg(test)]
 mod query {
-    use repository::mock::{mock_demographic_indicator_a, mock_program_a, MockDataInserts};
+    use repository::mock::{mock_immunisation_program, MockDataInserts};
```
Comment (jmbrunskill) [inline server/service/src/vaccine_course/test/query.rs:3]: small side track, but figured we should be using an immunisation program in our tests...

**Review by lache-melvin — APPROVED** [review]
Looks great James! I guess some of the vaccine course detail loaders can't be tested yet, as no data to query, but code looks good 🚀

File: server/graphql/types/src/types/vaccine_course_item.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+use async_graphql::*;
+
+use repository::vaccine_course::vaccine_course_item_row::VaccineCourseItemRow;
+
+#[derive(PartialEq, Debug)]
+pub struct VaccineCourseItemNode {
+    pub vaccine_course: VaccineCourseItemRow,
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply server/graphql/types/src/types/vaccine_course_item.rs:7]: Thanks, damn copy and paste bugs :)

**Pull request merged**